.travis.yml

```diff
@@ -6,8 +6,17 @@ before_script:
   - "npm install"
   - "npm run build"
   - "java -jar selenium-server-standalone-2.53.0.jar -Dphantomjs.binary.path=./node_modules/phantomjs-prebuilt/bin/phantomjs &"
-  - sleep 3
   - npm start &
-  - sleep 5
+  - "chmod 750 waiting-for-start.sh"
+  - "./waiting-for-start.sh"
+  - "STANDALONE=$(pgrep -n -f java)"
+  - "NODE=$(pgrep -n -f node)"
+  - "NPM=$(pgrep -n -f npm)"
```

Comment:
It is the case if running the program locally, but in travis ci, I need to
kill them both

2016-06-28 11:01 GMT-05:00 Xingchen Hong <notifications@github.com>:

> In .travis.yml
> <https://github.com/Zodiase/map-visualizer/pull/15#discussion_r68787219>:
>
> >    - npm start &
> > -  - sleep 5
> > +  - "chmod 750 waiting-for-start.sh"
> > +  - "./waiting-for-start.sh"
> > +  - "STANDALONE=$(pgrep -n -f java)"
> > +  - "NODE=$(pgrep -n -f node)"
> > +  - "NPM=$(pgrep -n -f npm)"
>
> I don't think you need to care npm. npm start runs node start and waits
> till node stops. Once you kill node, npm will end immediately.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Zodiase/map-visualizer/pull/15/files/1c3d4e5ef0ec58890c6b2a5a60b21090902a2c76#r68787219>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJZCbS-c4izAOGK5tV8p7hl2WYpQzpOdks5qQUVLgaJpZM4JAOXy>
> .
>